packages/eslint-plugin/src/rules/no-empty-interface.ts

```diff
@@ -59,6 +60,32 @@ export default util.createRule<Options, MessageIds>({
             context.report({
               node: node.id,
               messageId: 'noEmptyWithSuper',
+              fix(fixer) {
+                if (node.extends && node.extends.length) {
+                  const {
+                    id,
+                    extends: [extend],
+                    body,
+                  } = node;
+
+                  return [
+                    // replace `interface` keyword to `type`
+                    fixer.replaceTextRange(
+                      [id.parent?.range[0] || 0, id.range[0]],
+                      'type ',
+                    ),
+                    // replace `extends` keyword to `=` symbol
+                    fixer.replaceTextRange(
+                      [id.range[1], extend.range[0]],
+                      ' = ',
+                    ),
+                    // remove brackets `{}`
+                    fixer.removeRange([body.range[0] - 1, body.range[1]]),
+                  ];
```

Comment:
one thing that might be easier here, would be to replace the `node` text entirely, instead of selectively replacing pieces of it.

```ts
const sourceCode = context.getSourceCode(); // hoist this up into the root rule function.

fixer.replaceText(
  node,
  `type ${sourceCode.getText(node.id)} = ${sourceCode.getText(extends[0]};`,
);
```

Reply:
should be resolved now, thanks for a **nit**